avframe: Fix regression for differenceimageprovider that needs 32bit based pictures

cypline opened the pull request:
The internal memory is only 24 bit if using Format_RGB888, but the differenceproverider is relying on a 32 bit format.
This broke the difference view. This fix is repairing it considering the wish not to convert the picture twice.

cypline committed:
Update avframe.cpp

The internal memory is only 24 bit if using Format_RGB888, but the differenceproverider is relying on a 32 bit format.

bernhardu commented:
With this patch applied on top of 84c31e0f80 I get messages `QImage::scaled: Image is a null image` and the main picture stays empty. This is also with current Debian Testing.

cypline committed:
avframe: completed patch as with the changed format the scanline is different

The scanline of 3*w is not any longer valid if passed with 32 bit. So remove the parameter and use a different signature of the constructor.

cypline commented:
I found the reason, as I did the patches by hand on github directly, I oversaw a small change I did in line 132. Instead of QImage im(rgbbuffer, w, h, 3*w, QImage::Format_ARGB32, ::free, rgbbuffer); it must have been
QImage im(rgbbuffer, w, h, QImage::Format_ARGB32, ::free, rgbbuffer); I adopted this in my fork and committed it with
22f2582ec810f2a072c9dc547f5c0a2b64f3d6d6. Sorry for that convenience I introduced. I was to curious about the fixes to provide.

cypline commented:
With the patch you applied for differenceimageprovider which is the best in terms of speed and memory this and my other branch are obsolete.
